crates/eframe/src/web/events.rs

```diff
@@ -566,34 +566,22 @@ pub(crate) fn install_canvas_events(runner_ref: &WebRunner) -> Result<(), JsValu
                             let last_modified = std::time::UNIX_EPOCH
                                 + std::time::Duration::from_millis(file.last_modified() as u64);
 
-                            log::debug!("Loading {:?} ({} bytes)…", name, file.size());
-
-                            let future = wasm_bindgen_futures::JsFuture::from(file.array_buffer());
-
+                            log::debug!("File {:?} dropped", name);
                             let runner_ref = runner_ref.clone();
+                            // we use a future to access the runner_ref lock
                             let future = async move {
-                                match future.await {
-                                    Ok(array_buffer) => {
-                                        let bytes = js_sys::Uint8Array::new(&array_buffer).to_vec();
-                                        log::debug!("Loaded {:?} ({} bytes).", name, bytes.len());
-
-                                        if let Some(mut runner_lock) = runner_ref.try_lock() {
-                                            runner_lock.input.raw.dropped_files.push(
-                                                egui::DroppedFile {
-                                                    name,
-                                                    mime,
-                                                    last_modified: Some(last_modified),
-                                                    bytes: Some(bytes.into()),
-                                                    ..Default::default()
-                                                },
-                                            );
-                                            runner_lock.needs_repaint.repaint_asap();
-                                        }
-                                    }
-                                    Err(err) => {
-                                        log::error!("Failed to read file: {:?}", err);
-                                    }
-                                }
+                                if let Some(mut runner_lock) = runner_ref.try_lock() {
+                                    let stream_url =
+                                        web_sys::Url::create_object_url_with_blob(&file).ok();
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL_static this allocates an object that lives for the duration of `document`, meaning the memory will stay allocated:

>  for optimal performance and memory usage, if there are safe times when you can explicitly unload them, you should do so.

This feels like a new foot fgun: drag dropping many files will now slowly eat away your RAM

Reply:
So i guess i could revoke_object_url in get_data 

And also implement the Drop trait to drop the url ?

Reply:
Difficult.

One idea: `eframe::web` automatically calls `revoke_object_url` at the end of the frame unless the user has called `eframe::web::get_data`. This could be communicated with e.g. an `Arc<AtomicBool>`.